central/detection/service/service_impl.go

```diff
@@ -309,6 +326,19 @@ func (s *serviceImpl) DetectDeployTimeFromYAML(ctx context.Context, req *apiV1.D
 	}
 	eCtx.FetchOpt = fetchOpt
 
+	if req.GetCluster() != "" {
```

Comment:
Since we duplicate this, probably worthwhile to split this into a separate function similar to `getFetchOptionFromRequest`.

Reply:
I moved more of the 'duplicate' code into the util method - believe that's inline with the ask now: https://github.com/stackrox/stackrox/pull/7738/commits/177eeddbf27caa0c7f50eb031220982b05b04fe3